hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -203,30 +189,23 @@ public TokenRelationship loadPossiblyDeletedTokenRelationship(Token token, Accou
 	 * {@link TransactionRecordService} to update the {@link com.hedera.services.state.submerkle.ExpirableTxnRecord}
 	 * of the active transaction with these changes.
 	 *
-	 * @param tokenRelationships
-	 * 		the token relationships to save
+	 * @param tokenRelationships the token relationships to save
 	 */
-	public void persistTokenRelationships(final List<TokenRelationship> tokenRelationships) {
-		final var currentTokenRels = tokenRels.get();
-
+	public void commitTokenRelationships(final List<TokenRelationship> tokenRelationships) {
 		for (var tokenRelationship : tokenRelationships) {
 			final var key = EntityNumPair.fromModelRel(tokenRelationship);
 			if (tokenRelationship.isDestroyed()) {
-				currentTokenRels.remove(key);
-				backingTokenRels.removeFromExistingRels(legacyReprOf(tokenRelationship));
+				tokenRels.remove(Pair.of(tokenRelationship.getAccount().getId().asGrpcAccount(),
+						tokenRelationship.getToken().getId().asGrpcToken()));
 			} else {
-				persistNonDestroyed(tokenRelationship, key, currentTokenRels);
+				persistNonDestroyed(tokenRelationship, key, tokenRels);
 			}
 		}
 		sideEffectsTracker.trackTokenBalanceChanges(tokenRelationships);
 	}
 
 	private MerkleTokenRelStatus getMerkleTokenRelationship(Token token, Account account) {
-		final var tokenNum = EntityNum.fromModel(token.getId());
-		final var accountNum = EntityNum.fromModel(account.getId());
-		final var key = EntityNumPair.fromLongs(accountNum.longValue(), tokenNum.longValue());
-
-		return tokenRels.get().get(key);
+		return tokenRels.getRef(Pair.of(account.getId().asGrpcAccount(), token.getId().asGrpcToken()));
```

Comment:
Here we want `getImmutableRef()`, since we do **not** want to trigger a `getForModify()` in the `BackingStore`.